spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindFieldSelfAssignment.java

```diff
@@ -91,8 +93,12 @@ public void sawOpcode(int seen) {
                 OpcodeStack.Item fourth = stack.getStackItem(3);
                 XField f2 = third.getXField();
                 int registerNumber2 = fourth.getRegisterNumber();
-                if (f2 != null && f2.equals(getXFieldOperand()) && registerNumber2 >= 0
-                        && registerNumber2 == third.getFieldLoadedFromRegister()
+                int loadedFromRegister2 = fourth.getFieldLoadedFromRegister();
+                if (f2 != null && f2.equals(getXFieldOperand())
+                        && ((registerNumber2 >= 0 && registerNumber2 == third.getFieldLoadedFromRegister())
+                                || (loadedFromRegister2 >= 0 && third.getUserValue() != null
+                                        && third.getUserValue() instanceof Integer
+                                        && loadedFromRegister2 == (Integer) third.getUserValue()))
```

Comment:
Better to do not depends on auto boxing.
```suggestion
                                        && loadedFromRegister2 == ((Integer) third.getUserValue()).intValue()))
```